Sources/Datadog/DatadogCore/Context/DatadogContextProvider.swift

```diff
@@ -47,7 +47,7 @@ internal final class DatadogContextProvider {
     /// to make writes exclusive.
     private let queue = DispatchQueue(
         label: "com.datadoghq.core-context",
-        attributes: .concurrent
+        qos: .utility
```

Comment:
Given it is serial now, it no longer makes sense to do `.barrier` in following code, no?
```swift
func write(block: @escaping (inout DatadogContext) -> Void) {
   queue.async(flags: .barrier)
}
```

Reply:
Indeed, `.barrier` will have no effect on the serial queue, but I would like to keep it for now and move back to concurrency with a better threading design later.